File: module-contracts/src/main/java/com/beat/contracts/common/ReadModel.java
package com.beat.contracts.common;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Marks a module-contracts type as a read-only query result shape.
 *
 * <p>A ReadModel is not a domain model, JPA entity, or executable-module response DTO.
 * It exists only as an implementation-free query contract between an application query
 * service and an infra query adapter.</p>
 */
@Target(ElementType.TYPE)
@Retention(RetentionPolicy.RUNTIME)
public @interface ReadModel {
}

File: module-contracts/src/main/kotlin/com/beat/contracts/common/ReadModel.kt
package com.beat.contracts.common

/**
 * Marks a module-contracts type as a read-only query result shape.
 *
 * A ReadModel is not a domain model, JPA entity, or executable-module response DTO.
 * It exists only as an implementation-free query contract between an application query
 * service and an infra query adapter.
 */
@Target(AnnotationTarget.CLASS)
@Retention(AnnotationRetention.RUNTIME)
annotation class ReadModel
